### PR TITLE
refactor(versioning/pep440): range.ts

### PR DESCRIPTION
**Pull request by Gabriel-Ladzaretti**
## Changes:
refactored lib/versioning/pep440/range.ts.

## Context:
related to issue #6241

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:
- [x] Ran existing unit tests.
- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/versioning/pep440/range.ts [hunk 1]
```diff
@@ -193,3 +152,120 @@ export function isLessThanRange(input: string, range: string): boolean {
...
+    if (gte(newVersion, range.version)) {
+      // now here things get tricky
+      // we calculate the new future version
+      const futureVersion = getFutureVersion(range.version, newVersion, 1);
+      return range.operator + futureVersion;
+    }
+    // otherwise, treat it same as exclude
+    return range.operator + range.version;
+  }
+  // istanbul ignore next
```
Comment (viceice) [inline lib/versioning/pep440/range.ts:196]: When does this happen
?

File: lib/versioning/pep440/range.ts [hunk 2]
```diff
@@ -193,3 +152,120 @@ export function isLessThanRange(input: string, range: string): boolean {
...
+  currentVersion: string,
+  ranges: Range[]
+): string[] {
+  return ranges.map((range) => {
+    // bump lower bound to current new version
+    if (range.operator === '>=') {
+      return range.operator + newVersion;
+    }
+    return updateRangeValue(newVersion, currentValue, currentVersion, range);
+  });
```
Comment (viceice) [inline lib/versioning/pep440/range.ts:270]:
```suggestion
  }).filter(is.nonEmptyString);
```

**Review by viceice — CHANGES_REQUESTED** [review]
I'd this really a refactor PR or does it change runtime behavior?

**Comment by Gabriel-Ladzaretti** [comment]
this is purely a refactor PR, it shouldn't change any run time behavior.
Did this as a preparation to the needed changes required in #6241 that i want to implement.

**Comment by viceice** [comment]
> @viceice 
> This is purely a refactor PR, it shouldn't change any runtime behavior.
> Did this as a preparation to the required changes that I want to implement in issue #6241.

Ok, why the test then changed? Simply added test cases? Can't compare without spaces from mobile app.

**Review by viceice — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**